client/src/context/auth.test.js

```diff
@@ -0,0 +1,142 @@
+import { useAuth, AuthProvider } from "./auth";
+import axios from 'axios';
+import { render } from '@testing-library/react';
+
+//Mock dependencies that AuthContext uses
+jest.mock('axios');
+
+
+//Console log spy to observe child component that uses AuthContext
+let consoleLogSpy;
+
+
+//Test suite
+describe('AuthProvider Component', () => {
+
+    //Initialisation logic for all tests
+    beforeEach(() => {
+        jest.clearAllMocks();
+
+        //Mock localStorage window object
+        Object.defineProperty(window, 'localStorage', {
+            value: {
+                setItem: jest.fn(),
+                getItem: jest.fn(),
+                removeItem: jest.fn(),
+            },
+            writable: true,
+        });
+
+        //Reset headers
+        axios.defaults.headers.common = {};
+
+        //Reinitialise console log spy
+        consoleLogSpy = jest.spyOn(console, 'log').mockImplementation(() => {});
+    });
+
+
+    //Reset logic for all tests
+    afterEach(() => {
+        //Reset the getItem() method in localStorage mock after every test
+        localStorage.getItem.mockReset();
+
+        //Restore original functionality of console.log
+        consoleLogSpy.mockRestore();
+    });
+
+
+    //Default value for auth state
+    describe('should have default value for its auth state', () => {
+        beforeEach(() => {
```

Comment:
You have the same initialisation of consoleLogSpy in a beforeEach on line 34 as well. Not sure if this is intended.

Reply:
Thanks! It is indeed a duplicate call.